tasks/apm-inject-check.yml

```diff
@@ -0,0 +1,48 @@
+---
+- name: Fail if the Datadog Agent version is not compatible with APM host injection
+  fail:
+    msg: APM Host Injection is not supported for this datadog-agent version.
+  when: datadog_agent_major_version | int < 7
+
+- name: Fail if APM Host injection is not supported on this host
+  fail:
+    msg: APM Host Injection is not supported in this platform.
+  when: ansible_facts.os_family not in ["Debian", "RedHat", "Rocky", "AlmaLinux"]
+
+- name: Check if docker daemon config dir exists
+  stat:
+    path: /etc/docker
+  register: docker_daemon_config_dir
+
+- name: Fail if APM Host container injection requirements are met (Docker installed)
+  fail:
+    msg: /etc/docker does not exist. Please ensure docker is installed or disable the datadog_apm_host_docker_injection_enabled flag.
+  when: datadog_apm_host_docker_injection_enabled and not docker_daemon_config_dir.stat.isdir
```

Comment:
I just got this error while testing:

```
TASK [ansible-datadog : Fail if APM Host container injection requirements are met (Docker installed)] ************
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'datadog_apm_host_docker_injection_enabled and not docker_daemon_config_dir.stat.isdir' failed. The error was: error while evaluating conditional (datadog_apm_host_docker_injection_enabled and not docker_daemon_config_dir.stat.isdir): 'dict object' has no attribute 'isdir'\n\nThe error appears to be in '/home/slavek.kabrda/programming/ansible-datadog/tasks/apm-inject-check.yml': line 17, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Fail if APM Host container injection requirements are met (Docker installed)\n  ^ here\n"}
```

According to [stat docs](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/stat_module.html), you should do something like

```
when: datadog_apm_host_docker_injection_enabled and (docker_daemon_config_dir.stat.isdir is not defined or not docker_daemon_config_dir.stat.isdir)
```